pkg/controllers/disruption/helpers.go

```diff
@@ -191,6 +191,45 @@ func GetCandidates(ctx context.Context, cluster *state.Cluster, kubeClient clien
 	return lo.Filter(candidates, func(c *Candidate, _ int) bool { return shouldDeprovision(ctx, c) }), nil
 }
 
+// BuildDisruptionBudgets will return a map for nodePoolName -> numAllowedDisruptions and an error
+func BuildDisruptionBudgets(ctx context.Context, cluster *state.Cluster, clk clock.Clock, kubeClient client.Client) (map[string]int, error) {
```

Comment:
I wonder if it would simplify things to build a `state.budgets` struct that tracks this state (w/ state.cluster) as a member variable. You could write queries against this, and potentially keep errorable API calls out of the critical path.

Reply:
> You could write queries against this, and potentially keep errorable API calls out of the critical path

Reply:
Feel free to ignore if it doesn't make sense. This just set off my static helper smell detector.